src/canvas/Canvas.ts

```diff
@@ -24,6 +24,17 @@ import { TextEditingManager } from './TextEditingManager';
 
 const addEventOptions = { passive: false } as EventListenerOptions;
 
+const getEventPoints = (canvas: Canvas, e: TPointerEvent) => {
+  const viewportPoint = canvas.getViewportPoint(e);
+  const scenePoint = canvas.getScenePoint(e);
+  return {
+    viewportPoint,
+    scenePoint,
+    pointer: viewportPoint,
+    absolutePointer: scenePoint,
+  };
```

Comment:
there is anotherthing I have been thinking of lately
Since we cache the points an event can do
```
scenePoint.x += 5
```
And cause a bug all across fabric so we should return a clone
I will PR